File: src/main/java/com/thebluehats/server/core/modules/PerformanceStatsServiceModule.java
package com.thebluehats.server.core.modules;

import com.google.inject.AbstractModule;
import com.google.inject.Provides;
import com.google.inject.Singleton;
import com.thebluehats.server.api.daos.PerformanceStatsService;
import com.thebluehats.server.api.implementations.pitdata.PerformanceStatsServiceImpl;
import com.thebluehats.server.core.modules.annotations.ServerAPI;
import kong.unirest.UnirestInstance;

public class PerformanceStatsServiceModule extends AbstractModule {
    @Provides
    @Singleton
    static PerformanceStatsService provideGrindingSystem(@ServerAPI UnirestInstance serverAPI) {
        return new PerformanceStatsServiceImpl(serverAPI);
    }

    @Override
    protected void configure() {
    }
}

File: src/main/java/com/thebluehats/server/core/modules/PerformanceStatsServiceModule.kt
package com.thebluehats.server.core.modules

import com.google.inject.AbstractModule
import com.google.inject.Provides
import com.google.inject.Singleton
import com.thebluehats.server.api.daos.PerformanceStatsService
import com.thebluehats.server.api.implementations.pitdata.PerformanceStatsServiceImpl
import com.thebluehats.server.core.modules.annotations.ServerAPI
import kong.unirest.UnirestInstance

class PerformanceStatsServiceModule : AbstractModule() {
    override fun configure() {}

    companion object {
        @Provides
        @Singleton
        fun provideGrindingSystem(@ServerAPI serverAPI: UnirestInstance?): PerformanceStatsService {
            return PerformanceStatsServiceImpl(serverAPI!!)
        }
    }
}